src/actions/userActions.js

```diff
@@ -34,18 +54,17 @@ export const logout = () => async dispatch => {
     sessionService.deleteSession();
     sessionService.deleteUser();
     dispatch(logoutSuccess());
-  } catch (err) {
-    throw err;
+  } catch (error) {
+    dispatch(logoutError(error));
```

Comment:
I prefer just the throw. I think form errors should be managed at the component level.

Reply:
Throwing inside an action is a really bad pattern, since you are relying on it bubbling up to your component, and then either catching it inside an errorBoundary or a try/catch block.

Moreover, this is not a form error, it's an API error from the login request, so it should be handled by redux, same way we handle successful responses.